natlas-agent/natlas/screenshots.py

```diff
@@ -7,27 +7,86 @@
 import base64
 from urllib.parse import urlparse
 
+from PIL import Image, UnidentifiedImageError
+
 from natlas import logging
 from natlas import utils
 
 logger = logging.get_logger("ScreenshotUtils")
 
 
-def base64_image(path):
-    img = None
+def base64_file(path: str) -> str:
     with open(path, "rb") as f:
-        img = f.read()
-    return str(base64.b64encode(img))[2:-1]
+        return base64.b64encode(f.read()).decode("ascii")
+
+
+def is_valid_image(path: str) -> bool:
+    try:
+        Image.open(path)
+        return True
+    except (FileNotFoundError, UnidentifiedImageError):
+        return False
+
+
+def parse_url(url: str) -> tuple:
+    urlp = urlparse(url)
+    if not urlp.port:
+        port = 80 if urlp.scheme == "http" else 443
+    else:
+        port = urlp.port
+
+    return urlp.scheme.upper(), port
+
+
+def parse_aquatone_page(page: dict, file_path: str) -> dict:
+    if not (page["hasScreenshot"] and is_valid_image(file_path)):
+        return {}
+    scheme, port = parse_url(page["url"])
+    logger.info(f"{scheme} screenshot acquired for {page['hostname']} on port {port}")
+    return {
+        "host": page["hostname"],
+        "port": port,
+        "service": scheme,
+        "data": base64_file(file_path),
+    }
+
+
+def get_aquatone_session(base_dir: str) -> dict:
+    session_path = os.path.join(base_dir, "aquatone_session.json")
+    output = {}
+    if not os.path.isfile(session_path):
+        return output
```

Comment:
These two returns a bit weird. I'm not opposed, but I think returning output instead of returning {} makes it slightly harder to read. We don't even modify output so the name output incorrectly implies that it could be used to store and return a valid output.

In this case I would just return {} or rename it.